### ISSUE TITLE
Infinite Loop

### ISSUE THREAD

**Issue by CharlieTheProgrammer** [mcollina/autocannon#192]
The following code runs infinitely for some reason. Changing 'connections' to 1, resolves the issue.

```javascript
'use strict'

const http = require('http')
const autocannon = require("autocannon")
const server = http.createServer(handle)

server.listen(3000)

let count = 0
function handle (req, res) {
  console.log("Received Request #" + count)
  count++
  res.end('hello world')
}

function retrieve() {
  autocannon(
    {
      url: "http://localhost:3000",
      amount: 1,
      maxConnectionRequests: 1,
      maxOverallRequests: 2,
      connections: 3, //default
      pipelining: 1, // default
      duration: 1, // default
    },
    console.log
  )
}

retrieve()
```

**Comment by mcollina** [comment]
Good spot! Would you like to send a PR?

That combination of options is actually invalid, it would need to throw.

You cannot have amount < connections. Essentially we keep decreasing a counter waiting for a condition to trigger but that’s already in the past.

**Comment by thekemkid** [comment]
Closed via #193
